### PR TITLE
Update android-studio-canary to 3.0.0.17,171.4402976

### PR DESCRIPTION
**Pull request by ayatk**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/android-studio-canary.rb [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
   url "https://dl.google.com/dl/android/studio/ide-zips/#{version.before_comma}/android-studio-ide-#{version.after_comma}-mac.zip"
   name 'Android Studio Canary'
-  homepage 'https://sites.google.com/a/android.com/tools/download/studio/canary'
+  homepage 'https://developer.android.com/studio/preview/'
 
-  app "Android Studio #{version.major_minor}.app"
+  conflicts_with cask: 'android-studio'
+
+  app 'Android Studio.app'
```
Comment (commitay) [inline Casks/android-studio-canary.rb]: 😆  Guess your connection is faster than mine!

Comment (mxl) [reply Casks/android-studio-canary.rb]: @ayatk Why does it conflict with `android-studio`? It's not possible now to have `Android Studio` and `Android Studio Preview` installed at the same time.

Comment (commitay) [reply Casks/android-studio-canary.rb]: @mxl As of this version `android-studio` and `android-studio preview` both install `Android Studio.app`

Comment (mxl) [reply Casks/android-studio-canary.rb]: @commitay Why not use `app 'Android Studio.app', target: 'Android Studio Preview.app'`?

Comment (mxl) [reply Casks/android-studio-canary.rb]: #4656

**Pull request merged**